spanner/value.go

```diff
@@ -1844,6 +1850,50 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}, opts ...decodeO
 		*p = y
 	case *GenericColumnValue:
 		*p = GenericColumnValue{Type: t, Value: v}
+	case protoreflect.Enum:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if reflect.ValueOf(p).Kind() != reflect.Ptr {
+			return errNotAPointer(p)
+		}
+		if code != sppb.TypeCode_ENUM {
+			return errTypeMismatch(code, acode, ptr)
+		}
+		if isNull {
+			return errDstNotForNull(ptr)
+		}
+		x, err := getStringValue(v)
+		if err != nil {
+			return err
+		}
+		y, err := strconv.ParseInt(x, 10, 64)
+		if err != nil {
+			return errBadEncoding(v, err)
+		}
+		reflect.ValueOf(p).Elem().SetInt(y)
+	case proto.Message:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if reflect.ValueOf(p).Kind() != reflect.Ptr {
+			return errNotAPointer(p)
+		}
+		if code != sppb.TypeCode_PROTO {
```

Comment:
Different from the ENUM case, we do need to support decoding bytes typed result into PROTO. BYTES values and PROTO values are compatible with each other.

Please also add tests for both direction.

Reply:
@Shua1 
Supporting compatibility between BYTES and PROTO values involves code changes in the decoding function of BYTES also. This needs a bit more testing and handling corner cases.
I will make a note of this and will support this case in my subsequent PR. Does that sound good?

Reply:
@Shua1 
I have added the code changes and integration tests to validate the following compatibility behaviours between Bytes and Proto.

1. Write Bytes into Proto column
2. Write Proto into Bytes column
3. Read Proto column value using Bytes
4. Read Bytes column value using Proto